dirty_cat/similarity_encoder.py

```diff
@@ -267,6 +267,9 @@ def fit(self, X, y=None):
 
     def transform(self, X):
         """Transform X using specified encoding scheme.
+            For an n-gram similarity, the output dtype
+            will be the instance dtype, but the input dtype is
+            always float 32.
```

Comment:
I don't think that this is necessary: it's confusing for a user who does not look at the internals.